src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void growIfArrayIsFull() {
+        if (size == elements.length) {
+            int multiplalier = Math.round(elements.length * 1.5f);
+            System.arraycopy(elements, 0, elements = new Object[multiplalier], 0, size);
+        }
```

Comment:
The magic number '1.5f' should be replaced with a named constant to improve code readability.